local_pathfinding/obstacles.py

```diff
@@ -0,0 +1,340 @@
+            -the collision_cone represents the boats hull via a region of possible positions
+            -It is shaped like a cone, with the tip at the boat's position and the base at the
+             furthest possible position of the boat in the next time step
+            -the collision_cone is orientated according to the course of the boat
+            -position (XY): x,y coordinates of the boat's center (converted from lat/lon)
+            -course_over_ground (float): course of the boat in degrees, clockwise from true north
+            (#TODO: check if we need to consider magnetic north)
+            -width (float): width of the boat in meters
+            -length (float): length of the boat in meters
+    Notes
+    ----------
+        The following information about a boat is obtained from AIS:
+            - Maritime Mobile Service Identity (9 digit int)
+            - latitude (float) latitude in degrees
+            - longitude (float) longitude in degrees
+            - speed over ground (float) speed in knots over ground
+            - course over ground (float) degrees measured clockwise from true north
+            - boat dimensions (float) width and length of the boat in meters
+            - rate of turn (float) ROT in AISROT scale -126 to +126 corresponding to
+                -708 to +708 deg/min
+
+            - other information is also available but not used here
+
+        To avoid confusion, the physical dimensions, position, and COG of the boat are
+        stored in the aggregated Polygon object, not the Boat object itself.
+        So that all we need to do to update the boat's position or COG is to update the
+        polygon's position, size, and rotation.
+
+        points (x,y) are measured in km, with the origin set at the current global waypoint.
+
+    """
+
+    def __init__(
+        self,
+        reference: LatLon,
+        sailbot_position: LatLon,
+        sailbot_speed: float,
+        ais_ship: HelperAISShip,
+    ):
+        """
+        Args:
+            -reference (LatLon): latitude and longitude of the next global waypoint
+            -sailbot_position (LatLon): lat and lon position of SailBot
+            -sailbot_speed (float): speed of the SailBot in km/h
+            -ais_ship (HelperAISShip): an AISShip object, containing the following information:
+                -id (int): MMSI number of the boat
+                -width (float): width of the boat in meters
+                -length (float): length of the boat in meters
+                -lat_lon (LatLon): latitude and longitude of the boat
+                -speed_over_ground (float): speed of the boat in knots, over ground
+                -course_over_ground (float): COG of the boat, in degrees, clockwise from true north
+                -rate_of_turn (float): ROT of the boat in AISROT scale -126 to +126 corresponding
+                    to -708 to +708 degrees per minute
```

Comment:
Again no need to document here, people can refer to the definition of `HelperAISShip.msg`